addons/WritingCalculations/src/presenter.js

```diff
@@ -1053,6 +1078,25 @@ function AddonWritingCalculations_create() {
         });
     };
 
+    presenter.gradualShowAnswers = function (eventData) {
+        if (eventData.moduleID !== presenter.ID) {
+            return;
+        }
+
+        if (presenter.showAllAnswersInGSA) {
+            presenter.showAnswers();
```

Comment:
Jest to powtórzenie kodu + liczenie że nic się nie popsuje przy nowych zmianach.
Zrób proszę z tego oddzielną ścieżkę. Wspólne elementy wyodrębnij.

Obecnie wykonujesz presenter.showAnswers(); zanim zapiszesz wyniki. 2x wykonujesz disablowanie (jeden raz w showAnswers dla wszystkich gap, drugi raz tylko dla jednej gapy). W presenter.userAnswers masz duplikacje wartości. Masz dwa aktywne stany presenter.isShowAnswersActive i presenter.isGradualShowAnswersActive więc w nawiązaniu do mojego wcześniejszego komentarza to wykona się presenter.hideAnswers i presenter.gradualHideAnswers (o ile zastosujesz moją propozycję)